corehq/motech/repeaters/models.py

```diff
@@ -773,6 +781,14 @@ def cancel(self):
         self.next_check = None
         self.cancelled = True
 
+    def can_enqueue(self):
+        if self.next_check > datetime.utcnow():
+            # self is already in the queue, or it is not ready to be enqueued
+            return False
+        self.next_check += timedelta(hours=48)
+        self.save()
```

Comment:
Should catch `ResourceConfict` (and `return False`) here since two updates to the same record revision will cause that and we don't want it to bubble up. This is the crux of switching to optimistic locking and eliminating the old Redis locks.

Double-check that `ResourceConfict` is the exception raised when an old revision of a document is saved. I'm not positive above that.